## Zjazd 5 cz 1

In [3]:
import pandas as pd
import numpy as np

In [4]:
# Series z listy — domyślny indeks (0, 1, 2, ...)
oceny = pd.Series([4.5, 3.0, 5.0, 3.5, 4.0])
print(oceny)

0    4.5
1    3.0
2    5.0
3    3.5
4    4.0
dtype: float64


Zadanie 1a: Utwórz Series pensje z dict:

In [5]:
dane_pensje = {
    'Anna': 5500,
    'Jan': 7200,
    'Ewa': 4800,
    'Marek': 9100,
    'Kasia': 6300
}

pensje = pd.Series(dane_pensje)
print(pensje)

Anna     5500
Jan      7200
Ewa      4800
Marek    9100
Kasia    6300
dtype: int64


Zadanie 1b: Odpowiedz na pytania o pensje:

In [6]:
najbogatsza_osoba = pensje.idxmax()
print(f"Najwięcej zarabia: {najbogatsza_osoba}")

srednia = pensje.mean()
print(f"Średnia pensja wynosi: {srednia}")

powyzej_sredniej = pensje[pensje > srednia]
print("Osoby zarabiające powyżej średniej:")
print(powyzej_sredniej.index.tolist()) 

liczba_osob = (pensje < 6000).sum()
print(f"Liczba osób zarabiających poniżej 6000: {liczba_osob}")

Najwięcej zarabia: Marek
Średnia pensja wynosi: 6580.0
Osoby zarabiające powyżej średniej:
['Jan', 'Marek']
Liczba osób zarabiających poniżej 6000: 2


# Krok 2 — Operacje na Series
Zadanie 1c: Oblicz pensje po zmianach:

In [7]:
pensje_podwyzka = pensje * 1.10
print("Pensje po 10% podwyżce:")
print(pensje_podwyzka)

pensje_premia = pensje + 500
print("\nPensje po dodaniu 500 zł premii:")
print(pensje_premia)

nowa_pensja = (pensje * 1.10) + 500
print("\nNowa pensja (podwyżka + premia):")
print(nowa_pensja)

ranking = nowa_pensja.sort_values(ascending=False)
print("\nRanking nowych pensji (od najwyższej):")
print(ranking)

Pensje po 10% podwyżce:
Anna      6050.0
Jan       7920.0
Ewa       5280.0
Marek    10010.0
Kasia     6930.0
dtype: float64

Pensje po dodaniu 500 zł premii:
Anna     6000
Jan      7700
Ewa      5300
Marek    9600
Kasia    6800
dtype: int64

Nowa pensja (podwyżka + premia):
Anna      6550.0
Jan       8420.0
Ewa       5780.0
Marek    10510.0
Kasia     7430.0
dtype: float64

Ranking nowych pensji (od najwyższej):
Marek    10510.0
Jan       8420.0
Kasia     7430.0
Anna      6550.0
Ewa       5780.0
dtype: float64


# Ćwiczenie 2: DataFrame — tworzenie i atrybuty (20 min)
## Cel
Utwórz DataFrame z dict i wczytaj dane z pliku CSV.

## Krok 1 — DataFrame z dict

In [8]:
sklep = pd.DataFrame({
    'produkt': ['Laptop', 'Tablet', 'Smartfon', 'Słuchawki', 'Monitor'],
    'cena': [3500, 1800, 2500, 350, 2200],
    'stan_magazynowy': [45, 120, 200, 500, 75],
    'kategoria': ['Komputery', 'Mobilne', 'Mobilne', 'Akcesoria', 'Komputery']
})

Zadanie 2a: Zbadaj DataFrame sklep:

In [9]:
print("Kształt tabeli (wiersze, kolumny):")
print(sklep.shape)

print("\nNazwy kolumn:")
print(sklep.columns)

print("\nTypy danych:")
print(sklep.dtypes)

print("\nPierwsze 3 wiersze:")
print(sklep.head(3))

Kształt tabeli (wiersze, kolumny):
(5, 4)

Nazwy kolumn:
Index(['produkt', 'cena', 'stan_magazynowy', 'kategoria'], dtype='str')

Typy danych:
produkt              str
cena               int64
stan_magazynowy    int64
kategoria            str
dtype: object

Pierwsze 3 wiersze:
    produkt  cena  stan_magazynowy  kategoria
0    Laptop  3500               45  Komputery
1    Tablet  1800              120    Mobilne
2  Smartfon  2500              200    Mobilne


Zadanie 2b: Dodaj nowe kolumny do DataFrame sklep:

In [10]:
sklep['wartosc_magazynu'] = sklep['cena'] * sklep['stan_magazynowy']

sklep['cena_z_vat'] = sklep['cena'] * 1.23

print("Tabela sklep z nowymi kolumnami:")
print(sklep)

Tabela sklep z nowymi kolumnami:
     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0


## Krok 2 — Wczytanie CSV

In [11]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv'
tips = pd.read_csv(url)

Zadanie 2c: Podstawowa eksploracja datasetu tips:

In [12]:
print("Wymiary tabeli (wiersze, kolumny):")
print(sklep.shape)

print("\nOstatnie 5 wierszy:")
print(sklep.tail()) 

print("\nLosowe 3 wiersze:")
print(sklep.sample(3))

print("\nTypy danych w kolumnach:")
print(sklep.dtypes)

Wymiary tabeli (wiersze, kolumny):
(5, 6)

Ostatnie 5 wierszy:
     produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
0     Laptop  3500               45  Komputery            157500      4305.0
1     Tablet  1800              120    Mobilne            216000      2214.0
2   Smartfon  2500              200    Mobilne            500000      3075.0
3  Słuchawki   350              500  Akcesoria            175000       430.5
4    Monitor  2200               75  Komputery            165000      2706.0

Losowe 3 wiersze:
    produkt  cena  stan_magazynowy  kategoria  wartosc_magazynu  cena_z_vat
2  Smartfon  2500              200    Mobilne            500000      3075.0
4   Monitor  2200               75  Komputery            165000      2706.0
0    Laptop  3500               45  Komputery            157500      4305.0

Typy danych w kolumnach:
produkt                 str
cena                  int64
stan_magazynowy       int64
kategoria               str
wartosc_magazy

# Ćwiczenie 3: Eksploracja datasetu — samodzielna analiza (30 min)
## Cel
## Przeprowadź pełną eksplorację EDA na prawdziwym datasecie.


Dane

In [15]:
url = 'https://raw.githubusercontent.com/mwaskom/seaborn-data/master/penguins.csv'
penguins = pd.read_csv(url)


Zadanie 3a: Pierwszy rzut oka

In [17]:

print("Pierwsze 10 wierszy:")
print(penguins.head(10)) 

print("\nWymiary tabeli:")
print(penguins.shape)     

print("\nSzczegółowe informacje o zbiorze:")
penguins.info()

Pierwsze 10 wierszy:
  species     island  bill_length_mm  bill_depth_mm  flipper_length_mm  \
0  Adelie  Torgersen            39.1           18.7              181.0   
1  Adelie  Torgersen            39.5           17.4              186.0   
2  Adelie  Torgersen            40.3           18.0              195.0   
3  Adelie  Torgersen             NaN            NaN                NaN   
4  Adelie  Torgersen            36.7           19.3              193.0   
5  Adelie  Torgersen            39.3           20.6              190.0   
6  Adelie  Torgersen            38.9           17.8              181.0   
7  Adelie  Torgersen            39.2           19.6              195.0   
8  Adelie  Torgersen            34.1           18.1              193.0   
9  Adelie  Torgersen            42.0           20.2              190.0   

   body_mass_g     sex  
0       3750.0    MALE  
1       3800.0  FEMALE  
2       3250.0  FEMALE  
3          NaN     NaN  
4       3450.0  FEMALE  
5       3650.0

Zadanie 3b: Statystyki opisowe

In [18]:
print("Podstawowe statystyki (tylko numeryczne):")
display(penguins.describe())

print("\nStatystyki dla wszystkich kolumn (w tym tekstowych):")
display(penguins.describe(include='all'))

mediana_masy = penguins['body_mass_g'].median()
print(f"\nMediana masy ciała wynosi: {mediana_masy} g")

Podstawowe statystyki (tylko numeryczne):


,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g
count,342.000000,342.000000,342.000000,342.000000
mean,43.921930,17.151170,200.915205,4201.754386
std,5.459584,1.974793,14.061714,801.954536
min,32.100000,13.100000,172.000000,2700.000000
25%,39.225000,15.600000,190.000000,3550.000000
50%,44.450000,17.300000,197.000000,4050.000000
75%,48.500000,18.700000,213.000000,4750.000000
max,59.600000,21.500000,231.000000,6300.000000



Statystyki dla wszystkich kolumn (w tym tekstowych):


,species,island,bill_length_mm,bill_depth_mm,flipper_length_mm,body_mass_g,sex
count,344,344,342.000000,342.000000,342.000000,342.000000,333
unique,3,3,NaN,NaN,NaN,NaN,2
top,Adelie,Biscoe,NaN,NaN,NaN,NaN,MALE
freq,152,168,NaN,NaN,NaN,NaN,168
mean,NaN,NaN,43.921930,17.151170,200.915205,4201.754386,NaN
std,NaN,NaN,5.459584,1.974793,14.061714,801.954536,NaN
min,NaN,NaN,32.100000,13.100000,172.000000,2700.000000,NaN
25%,NaN,NaN,39.225000,15.600000,190.000000,3550.000000,NaN
50%,NaN,NaN,44.450000,17.300000,197.000000,4050.000000,NaN
75%,NaN,NaN,48.500000,18.700000,213.000000,4750.000000,NaN



Mediana masy ciała wynosi: 4050.0 g


Zadanie 3c: Rozkłady wartości

In [19]:

print("Liczba pingwinów według gatunku:")
print(penguins['species'].value_counts())

print("\nLiczba pingwinów na każdej wyspie:")
print(penguins['island'].value_counts())

print("\nRozkład płci:")
print(penguins['sex'].value_counts())

print("\nProcentowy udział gatunków:")
procenty = penguins['species'].value_counts(normalize=True) * 100
print(procenty)

Liczba pingwinów według gatunku:
species
Adelie       152
Gentoo       124
Chinstrap     68
Name: count, dtype: int64

Liczba pingwinów na każdej wyspie:
island
Biscoe       168
Dream        124
Torgersen     52
Name: count, dtype: int64

Rozkład płci:
sex
MALE      168
FEMALE    165
Name: count, dtype: int64

Procentowy udział gatunków:
species
Adelie       44.186047
Gentoo       36.046512
Chinstrap    19.767442
Name: proportion, dtype: float64


Zadanie 3d: Brakujące dane

In [20]:

braki_suma = penguins.isna().sum()
print("Liczba brakujących wartości w poszczególnych kolumnach:")
print(braki_suma)

braki_sex = penguins['sex'].isna().sum()
procent_brakow = (braki_sex / len(penguins)) * 100

print(f"\nW kolumnie 'sex' brakuje {procent_brakow:.2f}% danych.")

Liczba brakujących wartości w poszczególnych kolumnach:
species               0
island                0
bill_length_mm        2
bill_depth_mm         2
flipper_length_mm     2
body_mass_g           2
sex                  11
dtype: int64

W kolumnie 'sex' brakuje 3.20% danych.


Zadanie 3e: Wnioski (refleksja) W komórce Markdown napisz minimum 5 obserwacji o datasecie. Każda obserwacja to jedno zdanie poparte liczbą z twoich obliczeń powyżej. Przykłady:

Liczba danych: Dataset zawiera 344 pingwiny opisane przez 7 różnych cech (kolumn).

Gatunki: Mamy 3 gatunki pingwinów. Najczęstszy to Adelie (152 osobniki, co stanowi ok. 44.19% całego zbioru).

Brakujące dane:  W kolumnach dotyczących wymiarów fizycznych (dziób, skrzydło, masa) brakuje po 2 wartości.

Najwięcej braków jest w kolumnie sex – brakuje tam danych dla 11 pingwinów (3.20% zbioru).

Masa ciała:  Średnia masa ciała to ok. 4201.75 g.

Mediana wynosi dokładnie 4050.0 g.

Rozstęp jest spory: najlżejszy pingwin waży 2700 g, a najcięższy aż 6300 g.

Lokalizacja: Pingwiny mieszkają na 3 wyspach. Najliczniejszą populację znajdziemy na wyspie Biscoe (168 osobników).

# Ćwiczenie 4: Pytania biznesowe + commit (15 min)
Cel

Odpowiedz na pytania biznesowe o danych i zapisz pracę w Git.

Dane — wróć do datasetu tips

In [21]:
tips = pd.read_csv('https://raw.githubusercontent.com/mwaskom/seaborn-data/master/tips.csv')

# Dodaj kolumnę z procentem napiwku
tips['tip_pct'] = (tips['tip'] / tips['total_bill'] * 100).round(1)

Zadanie 4a:

In [22]:

sredni_rachunek = tips['total_bill'].mean()
print(f"Średni rachunek: {sredni_rachunek:.2f} $")

sredni_procent_napiwku = tips['tip_pct'].mean()
print(f"Średni napiwek w %: {sredni_procent_napiwku:.1f}%")

print("\nLiczba rachunków według dni tygodnia:")
print(tips['day'].value_counts())

liczba_palaczy = tips['smoker'].value_counts()
print("\nPodział na palaczy i niepalących:")
print(liczba_palaczy)

print("\nStatystyki dla kolumny 'tip':")
print(tips['tip'].describe())

rozklad_posilkow = tips['time'].value_counts(normalize=True) * 100
print(f"\nProcentowy udział posiłków:\n{rozklad_posilkow}")

unikalne_rozmiary = tips['size'].nunique()
print(f"\nLiczba unikalnych rozmiarów grup (stolików): {unikalne_rozmiary}")
sredni_rachunek = tips['total_bill'].mean()
print(f"Średni rachunek: {sredni_rachunek:.2f} $")

sredni_procent_napiwku = tips['tip_pct'].mean()
print(f"Średni napiwek w %: {sredni_procent_napiwku:.1f}%")

print("\nLiczba rachunków według dni tygodnia:")
print(tips['day'].value_counts())

liczba_palaczy = tips['smoker'].value_counts()
print("\nPodział na palaczy i niepalących:")
print(liczba_palaczy)

print("\nStatystyki dla kolumny 'tip':")
print(tips['tip'].describe())

rozklad_posilkow = tips['time'].value_counts(normalize=True) * 100
print(f"\nProcentowy udział posiłków:\n{rozklad_posilkow}")

unikalne_rozmiary = tips['size'].nunique()
print(f"\nLiczba unikalnych rozmiarów grup (stolików): {unikalne_rozmiary}")

Średni rachunek: 19.79 $
Średni napiwek w %: 16.1%

Liczba rachunków według dni tygodnia:
day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64

Podział na palaczy i niepalących:
smoker
No     151
Yes     93
Name: count, dtype: int64

Statystyki dla kolumny 'tip':
count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%        2.000000
50%        2.900000
75%        3.562500
max       10.000000
Name: tip, dtype: float64

Procentowy udział posiłków:
time
Dinner    72.131148
Lunch     27.868852
Name: proportion, dtype: float64

Liczba unikalnych rozmiarów grup (stolików): 6
Średni rachunek: 19.79 $
Średni napiwek w %: 16.1%

Liczba rachunków według dni tygodnia:
day
Sat     87
Sun     76
Thur    62
Fri     19
Name: count, dtype: int64

Podział na palaczy i niepalących:
smoker
No     151
Yes     93
Name: count, dtype: int64

Statystyki dla kolumny 'tip':
count    244.000000
mean       2.998279
std        1.383638
min        1.000000
25%      